### PR TITLE
[SPARK-39857][SQL][TESTS][FOLLOW-UP] Make "translate complex expression" pass with ANSI mode on

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR fixes `translate complex expression` to pass with ANSI mode on. We do push `Abs` with ANSI mode on (https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala#L93): 

```
[info] - translate complex expression *** FAILED *** (22 milliseconds)
[info]   Expected None, but got Some((ABS(cint) - 2) <= 1) (DataSourceV2StrategySuite.scala:325)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
[info]   at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:471)
[info]   at org.scalatest.funsuite.AnyFunSuite.newAssertionFailedException(AnyFunSuite.scala:1563)
[info]   at org.scalatest.Assertions.assertResult(Assertions.scala:867)
[info]   at org.scalatest.Assertions.assertResult$(Assertions.scala:863)
[info]   at org.scalatest.funsuite.AnyFunSuite.assertResult(AnyFunSuite.scala:1563)
[info]   at org.apache.spark.sql.execution.datasources.v2.DataSourceV2StrategySuite.testTranslateFilter(DataSourceV2StrategySuite.scala:325)
[info]   at org.apache.spark.sql.execution.datasources.v2.DataSourceV2StrategySuite.$anonfun$new$4(DataSourceV2StrategySuite.scala:176)
[info]   at org.apache.spark.sql.execution.datasources.v2.DataSourceV2StrategySuite.$anonfun$new$4$adapted(DataSourceV2StrategySuite.scala:170)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at org.apache.spark.sql.execution.datasources.v2.DataSourceV2StrategySuite.$anonfun$new$3(DataSourceV2StrategySuite.scala:170)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:190)
[info]   at org.apache.spark.SparkFunSuite.withFixture(SparkFunSuite.scala:204)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:188)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:200)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:200)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:182)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterEach$$super$runTest(SparkFunSuite.scala:65)
[info]   at org.scalatest.BeforeAndAfterEach.runTest(BeforeAndAfterEach.scala:234)
[info]   at org.scalatest.BeforeAndAfterEach.runTest$(BeforeAndAfterEach.scala:227)
[info]   at org.apache.spark.SparkFunSuite.runTest(SparkFunSuite.scala:65)
```

https://github.com/apache/spark/runs/7595362617?check_suite_focus=true

### Why are the changes needed?

To make the build pass with ANSI mode on.

### Does this PR introduce _any_ user-facing change?

No, test-only.

### How was this patch tested?

Manually ran the unittest with ANSI mode on.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala [hunk 1]
```diff
@@ -173,7 +173,7 @@ class DataSourceV2StrategySuite extends PlanTest with SharedSparkSession {
       testTranslateFilter(LessThanOrEqual(
         // Expressions are not supported
         // Functions such as 'Abs' are not supported
-        Subtract(Abs(attrInt), 2), 1), None)
+        Subtract(Abs(attrInt, failOnError = false), 2), 1), None)
```
Comment (huaxingao) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala:176]: nit: for the above comment, shall we be more specific and say 
```
// Functions such as 'Abs' are not pushed down with ANSI mode off
```

**Review by huaxingao — APPROVED** [review]

**Comment by huaxingao** [comment]
@HyukjinKwon Thanks for catching the problem! I will back port the fix in 3.3

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala [hunk 2]
```diff
@@ -173,7 +173,7 @@ class DataSourceV2StrategySuite extends PlanTest with SharedSparkSession {
       testTranslateFilter(LessThanOrEqual(
         // Expressions are not supported
         // Functions such as 'Abs' are not supported
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala:175]:
```suggestion
        // Functions such as 'Abs' are not pushed down with ANSI mode off
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala [hunk 3]
```diff
@@ -200,7 +200,7 @@ class DataSourceV2StrategySuite extends PlanTest with SharedSparkSession {
         And(
           GreaterThan(attrInt, 1),
           // Functions such as 'Abs' are not supported
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala:202]:
```suggestion
          // Functions such as 'Abs' are not pushed down with ANSI mode off
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala [hunk 4]
```diff
@@ -211,7 +211,7 @@ class DataSourceV2StrategySuite extends PlanTest with SharedSparkSession {
         Or(
           LessThanOrEqual(attrInt, 1),
           // Functions such as 'Abs' are not supported
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala:213]:
```suggestion
          // Functions such as 'Abs' are not pushed down with ANSI mode off
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala [hunk 5]
```diff
@@ -241,7 +241,7 @@ class DataSourceV2StrategySuite extends PlanTest with SharedSparkSession {
         Or(
           EqualTo(attrInt, 1),
           // Functions such as 'Abs' are not supported
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala:243]:
```suggestion
          // Functions such as 'Abs' are not pushed down with ANSI mode off
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala [hunk 6]
```diff
@@ -277,7 +277,7 @@ class DataSourceV2StrategySuite extends PlanTest with SharedSparkSession {
         ),
         And(
           // Functions such as 'Abs' are not supported
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala:279]:
```suggestion
          // Functions such as 'Abs' are not pushed down with ANSI mode off
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala [hunk 7]
```diff
@@ -307,7 +307,7 @@ class DataSourceV2StrategySuite extends PlanTest with SharedSparkSession {
         ),
         Or(
           // Functions such as 'Abs' are not supported
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2StrategySuite.scala:309]:
```suggestion
          // Functions such as 'Abs' are not pushed down with ANSI mode off
```

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.3.

**Comment by huaxingao** [comment]
Thanks @HyukjinKwon

**Pull request closed without merging**